diff --git a/node_modules/react-sound/lib/index.js b/node_modules/react-sound/lib/index.js
index cb4696e..976cdf1 100644
--- a/node_modules/react-sound/lib/index.js
+++ b/node_modules/react-sound/lib/index.js
@@ -114,6 +114,8 @@ var Sound = function (_React$Component) {
   }, {
     key: 'updateSound',
     value: function updateSound(sound) {
+
+      console.log(`updateSound(${sound})`)
       var prevProps = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
 
       if (!sound) {
@@ -163,6 +165,8 @@ var Sound = function (_React$Component) {
   }, {
     key: 'createSound',
     value: function createSound(callback) {
+
+      console.log(`createSound`)
       var _this4 = this;
 
       this.removeSound();
@@ -218,6 +222,8 @@ var Sound = function (_React$Component) {
   }, {
     key: 'removeSound',
     value: function removeSound() {
+
+      console.log(`removeSound`)
       if (this.stopCreatingSound) {
         this.stopCreatingSound();
         delete this.stopCreatingSound;
@@ -234,6 +240,8 @@ var Sound = function (_React$Component) {
   }, {
     key: 'render',
     value: function render() {
+
+      console.log(`render`)
       return null;
     }
   }]);
